### PR TITLE
Added test for default texture draw bug in Chrome

### PR DESCRIPTION
**Pull request by toji**
Issue uncovered by the OES_element_index_uint extension test, as described here: https://code.google.com/p/chromium/issues/detail?id=524144

### CODE REVIEW

File: sdk/tests/conformance/rendering/default-texture-draw-bug.html [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+** IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+** CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+** TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+** MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+-->
+
+<!DOCTYPE html>
```
Comment (kenrussell) [inline sdk/tests/conformance/rendering/default-texture-draw-bug.html]: We've removed this header from a bunch of Mobica's tests recently, so not sure we should cargo-cult its use forward. Your call.

File: sdk/tests/conformance/rendering/default-texture-draw-bug.html [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        quadArray[idx++] = 1.0;
+        quadArray[idx++] = 0.0;
+
+        quadArray[idx++] = -1.0;
+        quadArray[idx++] = -1.0;
+        quadArray[idx++] = 0.0;
+
+        quadArray[idx++] = 1.0;
+        quadArray[idx++] = -1.0;
+        quadArray[idx++] = 0.0;
```
Comment (kenrussell) [inline sdk/tests/conformance/rendering/default-texture-draw-bug.html]: Could you reuse WebGLTestUtils.setupQuad for this?

File: sdk/tests/conformance/rendering/default-texture-draw-bug.html [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+        var vertexObject = gl.createBuffer();
+        gl.bindBuffer(gl.ARRAY_BUFFER, vertexObject);
+        gl.bufferData(gl.ARRAY_BUFFER, quadArray, gl.STATIC_DRAW);
+
+        var indexObject = gl.createBuffer();
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, indexObject);
+        gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, new Uint16Array([0, 1, 2, 2, 3, 0]), gl.STATIC_DRAW);
```
Comment (kenrussell) [inline sdk/tests/conformance/rendering/default-texture-draw-bug.html]: Or, rather, setupIndexedQuad?

File: sdk/tests/conformance/rendering/default-texture-draw-bug.html [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        // Clear to white.
+        gl.clearColor(1.0, 1.0, 1.0, 1.0);
+        gl.clear(gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT);
+
+        // Draw without binding any textures.
+        gl.drawElements(gl.TRIANGLES, 6, gl.UNSIGNED_SHORT, 0);
+
+        // Check to ensure the entire canvas is black.
+        wtu.checkCanvasRect(gl, 0.0, 0.0, canvas.width, canvas.height,
+                [0.0, 0.0, 0.0], "Draw should pass", 2);
```
Comment (kenrussell) [inline sdk/tests/conformance/rendering/default-texture-draw-bug.html]: Wondering whether the canvas's size can be minimized to reduce the run time of the readPixels calls.

**Comment by kenrussell** [comment]
Excellent. Thanks for reducing this test case. Could you investigate the cleanups mentioned above? Please merge yourself afterward. Thanks.

File: sdk/tests/conformance/rendering/default-texture-draw-bug.html (hunk 4 above)
Comment (toji) [inline sdk/tests/conformance/rendering/default-texture-draw-bug.html]: Doing so seems to reduce the frequency with which the test reproduces the bug.

**Pull request merged**